web-wallet/src/lib/components/Balance/Balance.svelte

```diff
@@ -23,37 +26,100 @@
   /** @type {string} */
   export let tokenCurrency;
 
-  /** @type {number} */
-  export let tokens;
+  /** @type {bigint} */
+  export let shieldedAmount;
 
-  /** @type {number | undefined} */
-  export let shieldedTokensPercentage = undefined;
+  /** @type {bigint} */
+  export let unshieldedAmount;
+
+  $: totalBalance = luxToDusk(shieldedAmount) + luxToDusk(unshieldedAmount);
+
+  $: shieldedPercentage = (luxToDusk(shieldedAmount) / totalBalance) * 100;
+  $: unshieldedPercentage = (luxToDusk(unshieldedAmount) / totalBalance) * 100;
 
   $: classes = makeClassName(["dusk-balance", className]);
+
   $: duskFormatter = createCurrencyFormatter(locale, tokenCurrency, 9);
   $: fiatFormatter = createCurrencyFormatter(locale, fiatCurrency, 2);
+  $: numberFormatter = createNumberFormatter(locale, 2);
```

Comment:
The formatter stuff is a mess, we need to refactor this, once we have some free time.

https://github.com/dusk-network/rusk/issues/1316